File: oj/sdufe/contest/2020-culture/p20206.cpp
#include <algorithm>
#include <cmath>
#include <cstdio>
#include <cstring>
#include <iostream>
#include <map>
#include <queue>
#include <set>
#include <string>
#include <vector>
#define _for(i, init, lim) for (int i = (init); i < (lim); i++)
#define _rep(i, init, lim_eq) for (int i = (init); i <= (lim_eq); ++i)
int T, n, m;
int num_c;
int pos(int r, int c) {
    int filp = c % 2 == 0;
    int idx = (c - 1) * n;
    idx += (filp ? n - r + 1 : r);
    return idx;
}
int main(int argc, char const* argv[]) {
    std::ios::sync_with_stdio(false);
    std::cin >> T;
    while (T--) {
        std::cin >> n >> m;
        num_c = ceil(1.0 * m / n);
        _rep(r, 1, n) {
            _rep(c, 1, num_c) {
                int idx = pos(r, c);
                if (idx > m) {
                    break;
                }
                std::cout << idx;
                if (pos(r, c + 1) <= m) {
                    std::cout << ' ';
                }
            }
            std::cout << std::endl;
        }
    }
    return 0;
}

File: oj/sdufe/contest/2020-culture/p20206.py
import math

global T, m, n


def idx(x, y) -> int:
    filp = x % 2 == 0
    pos = (x - 1) * n
    pos += n - y + 1 if filp else y
    return pos


if __name__ == '__main__':
    T = int(input())
    while T != 0:
        n, m = map(int, input().split())
        numC = math.ceil(m/n)
        for r in range(1, n+1):
            for c in range(1, numC+1):
                id = idx(c, r)
                if id > m:
                    break
                print(id, end=' ')
            print()
        T -= 1